File: java-spring/crud/src/main/java/com/example/crud/service/UpdateFeedService.java
package com.example.crud.service;

import com.example.crud.controller.dto.request.UpdateFeedRequest;
import com.example.crud.entity.Feed;
import com.example.crud.repository.FeedRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@RequiredArgsConstructor
@Service
public class UpdateFeedService {

    private final FeedRepository feedRepository;

    @Transactional
    public void execute(Long id, UpdateFeedRequest request) {

        Feed feed = feedRepository.findById(id)
                .orElseThrow(RuntimeException::new);

        feed.update(request.getTitle(), request.getContent());
    }
}

File: kotlin-spring/crud/src/main/kotlin/com/example/crud/service/UpdateFeedService.kt
package com.example.crud.service

import com.example.crud.controller.dto.request.UpdateFeedRequest
import com.example.crud.entity.Feed
import com.example.crud.repository.FeedRepository
import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Service
import kotlin.RuntimeException

@Service
class UpdateFeedService(
        private val feedRepository: FeedRepository,
) {

    fun execute(feedId: Long, request: UpdateFeedRequest) {

        val feed: Feed = feedRepository.findByIdOrNull(feedId) ?: throw RuntimeException()

        feed.update(
                title = request.title,
                content = request.content,
        )
    }
}